src/realm/util/file.cpp

```diff
@@ -680,20 +680,50 @@ void File::resize(SizeType size)
 }
 
 
-void File::prealloc(SizeType offset, size_t size)
+void File::prealloc(size_t size)
 {
     REALM_ASSERT_RELEASE(is_attached());
 
 #if defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 200112L // POSIX.1-2001 version
 
-    prealloc_if_supported(offset, size);
+    prealloc_if_supported(0, size);
 
 #else // Non-atomic fallback
 
-    if (int_add_with_overflow_detect(offset, size))
-        throw std::runtime_error("File size overflow");
-    if (get_size() < offset)
-        resize(offset);
+#if REALM_IOS
+    fstore_t store = { F_ALLOCATEALL, F_PEOFPOSMODE, 0, static_cast<off_t>(size), 0 };
+    int ret = fcntl(m_fd, F_PREALLOCATE, &store);
+    if (ret == -1) {
+        int err = errno;
+        std::string msg = get_errno_msg("fcntl() inside prealloc() failed: ", err);
+    }
+    ret = ftruncate(m_fd, size);
+    if (ret != 0) {
+        int err = errno;
+        std::string msg = get_errno_msg("ftruncate() inside prealloc() failed: ", err);
+    }
+#elif REALM_ANDROID || defined(_WIN32)
+    if (size > get_size()) {
+        size_t new_size = size;
+        if (m_encryption_key) {
+            new_size = static_cast<size_t>(data_size_to_encrypted_size(size));
+            if (new_size < size) {
+                throw std::runtime_error("File size overflow");
```

Comment:
It might be useful to include both sizes in the error message

Reply:
Not just "might". We have to :-)
With errors, always, always include as much info as possible.